src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -255,7 +255,7 @@ static ModelTypeTable createTypeTable() {
 
   @VisibleForTesting
   List<StaticLangApiMethodView> generateApiMethods(
-      InterfaceContext context, Iterable<MethodModel> methods) {
+      InterfaceContext context, Iterable<? extends MethodModel> methods) {
```

Comment:
Does this `GoGapicSufaceTransformer` apply to both `Gapic` and `Discovery` APIs? If so this method LGTM. Otherwise (if GapicSurfaceTransformer works for Gapics only), it might be helpful to restrict the type of `methods` to `Iterable<ProtoMethodModel>` or `Iterable<? extends ProtoMethodModel>`

Reply:
What is the added value of providing the concrete class implementation over the base abstract class?

Reply:
It may help to error out earlier and help debugging. (E.g., if a DiscoMethodModel is passed in accidentally, the code will fail at compile time rather than runtime)

I mean if discogapics can use the same concrete `SurfaceTransformer` as gapics, this is fine with me. Otherwise, keeping accessibility as strict as possible should be a good practice for Java.

Reply:
Ideally the discogapics would use the same concrete SurfaceTransformer class as gapics, in the way JavaSurfaceTransformer supports both formats.

I agree that keeping accessibility strict helps with debugging and preventing bugs, but as long as we're having discussions about discogapics for Go, I would like to prevent devs from using `Gapic*` specific methods and entities in this SurfaceTransformer class, so that it makes it easier to reuse the existing abstractions.